packages/cortex-react/src/tests/Calendar.test.tsx

```diff
@@ -0,0 +1,34 @@
+import '@testing-library/jest-dom';
+import { render, renderHook, screen } from '@testing-library/react';
+import React from 'react';
+import { Calendar } from '../components/Calendar';
+import { useCalendar } from '../hooks';
+
+jest.mock('../components', () => ({
+  CalendarHeader: () => <div data-testid="calendar-header" />,
+  CalendarGrid: () => <div data-testid="calendar-grid" />,
+}));
+
+describe('Calendar', () => {
+  it('Should render', () => {
+    const value = new Date();
+    const onChange = () => undefined;
+
+    renderHook(() => {
+      return useCalendar({
+        value,
+        onChange,
+      });
+    });
+
+    render(<Calendar value={value} onChange={onChange} />);
+
+    const calendarElement = screen.getByTestId('calendar-div');
+    const calendarHeaderElement = screen.getByTestId('calendar-header');
+    const calendarGridElement = screen.getByTestId('calendar-grid');
+
+    expect(calendarElement).toBeInTheDocument();
+    expect(calendarHeaderElement).toBeInTheDocument();
+    expect(calendarGridElement).toBeInTheDocument();
```

Comment:
Nesse componente de <Calendar/> dá pra testar seleção de data?

Reply:
Eu acho que o teste precisaria ser end to end ou algo do gênero para testar, pois precisaria criar estado e ver se ta setando certo no clique. Ao mesmo tempo, o clique em si acontece la no CalendarCell e acho que quando fosse rodar ia precisar de vários mocks. Por isso acabei testando todos outros componentes e hooks com mais cuidado. O calendar no fim só renderiza dois componentes, mas não tem nenhuma grande lógica envolvida.

Ou seja, não fui capaz kk😢